Makefile

```
@@ -1,199 +1,173 @@
+  # The problem is that we want to be able to have targets like F3DEX2_2.08,
+  # but this would also be the directory itself, whose existence and possible
+  # modification needs to be handled by the Makefile. It is possible to write
+  # the Makefile where the directory is the main target for that microcode, but
+  # this has worse behavior in case of modification to the directory. Worse, if
+  # it was done this way, then it would break if the user tried to set
+  # PARENT_OUTPUT_DIR anywhere else. So, better to support building everywhere
+  # but here than to support only building here.
+endif
+
+NO_COL := \033[0m
+RED    := \033[0;31m
+GREEN  := \033[0;32m
+YELLOW := \033[0;33m
+BLUE   := \033[0;34m
+INFO    := $(BLUE)
+SUCCESS := $(GREEN)
+FAILURE := $(RED)
+WARNING := $(YELLOW)
+
+$(PARENT_OUTPUT_DIR):
+	@printf "$(INFO)Creating parent output directory$(NO_COL)\n"
+ifeq ($(OS),Windows_NT)
+	mkdir $@
+else
+	mkdir -p $@
+endif
+ALL_UCODES :=
+ALL_UCODES_WITH_MD5S :=
+ALL_OUTPUT_DIRS :=
+
+define reset_vars
+  NAME := 
+  DESCRIPTION := (Not used in any retail games)
+  ID_STR := Custom F3DEX2-based microcode, github.com/Mr-Wiseguy/f3dex2 & Nintendo
+  ID_STR := RSP Gfx ucode F3DEX       fifo 2.04H Yoshitaka Yasumoto 1998 Nintendo.
+  MD5_CODE := 
+  MD5_DATA := 
+  OPTIONS := 
+  EXTRA_DEPS :=
+endef
+
+define ucode_rule
+  # Variables defined outside the function need one dollar sign, whereas
+  # variables defined within the function need two. This is because make first
+  # expands all this text, substituting single-dollar-sign variables, and then
+  # executes all of it, causing all the assignments to actually happen.
+  ifeq ($(NAME),)
+   $$(error Microcode name not set!)
+  endif
+  UCODE_OUTPUT_DIR := $(PARENT_OUTPUT_DIR)/$(NAME)
+  CODE_FILE := $$(UCODE_OUTPUT_DIR)/$(NAME).code
+  DATA_FILE := $$(UCODE_OUTPUT_DIR)/$(NAME).data
+  SYM_FILE  := $$(UCODE_OUTPUT_DIR)/$(NAME).sym
+  TEMP_FILE := $$(UCODE_OUTPUT_DIR)/$(NAME).tmp.s
+  ALL_UCODES += $(NAME)
+  ifneq ($(MD5_CODE),)
+   ALL_UCODES_WITH_MD5S += $(NAME)
+  endif
+  ALL_OUTPUT_DIRS += $$(UCODE_OUTPUT_DIR)
+  OFF_OPTIONS := $(filter-out $(OPTIONS),$(ALL_OPTIONS))
+  OPTIONS_EQU := 
+  $$(foreach option,$(OPTIONS),$$(eval OPTIONS_EQU += -equ $$(option) 1))
+  OFF_OPTIONS_EQU := 
+  $$(foreach o2,$$(OFF_OPTIONS),$$(eval OFF_OPTIONS_EQU += -equ $$(o2) 0))
+  ARMIPS_CMDLINE := \
+   -strequ CODE_FILE $$(CODE_FILE) \
+   -strequ DATA_FILE $$(DATA_FILE) \
+   $$(OPTIONS_EQU) \
+   $$(OFF_OPTIONS_EQU) \
+   f3dex2.s \
+   -sym2 $$(SYM_FILE) \
+   -temp $$(TEMP_FILE)
+  # Microcode target
+  .PHONY: $(NAME)
+  $(NAME): $$(CODE_FILE)
+  # Directory target variables, see below.
+  $$(UCODE_OUTPUT_DIR): UCODE_OUTPUT_DIR:=$$(UCODE_OUTPUT_DIR)
+  # Directory target recipe
+  $$(UCODE_OUTPUT_DIR):
+	@echo "$(INFO)Creating directory $$(UCODE_OUTPUT_DIR)$(NO_COL)"
+  ifeq ($(OS),Windows_NT)
+	@mkdir $$(UCODE_OUTPUT_DIR)
```

I missed trying to build from a clean state where directories had to be made, so the subst is still needed on Windows. However since it breaks the md5sum command, it has to be done here instead of how it was before. I've tested that this change fixes it.
```suggestion
	@mkdir $$(subst /,\,$$(UCODE_OUTPUT_DIR))
```